app/gui/view/graph-editor/src/builtin/visualization/native/text_visualization/text_provider.rs

```diff
@@ -296,6 +940,18 @@ impl TryFrom<visualization::Data> for LazyGridDataUpdate {
     }
 }
 
+impl TryFrom<serde_json::Value> for LazyGridData {
+    type Error = visualization::DataError;
+
+    fn try_from(data: serde_json::Value) -> Result<Self, Self::Error> {
+        Ok(serde_json::from_value(data.clone()).unwrap_or_else(|_| {
+            let data_str = serde_json::to_string_pretty(&data);
+            let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {e}.>"));
+            data_str.into()
+        }))
+    }
```

Comment:
Why return `Ok` if the input can't be deserialized?

Reply:
The whole implementation is actually no longer needed. Removed it.